samza-core/src/main/scala/org/apache/samza/job/local/ThreadJobFactory.scala

```diff
@@ -45,20 +47,36 @@ import scala.collection.mutable
   * Creates a new Thread job with the given config
   */
 class ThreadJobFactory extends StreamJobFactory with Logging {
-  def getJob(config: Config): StreamJob = {
+  def getJob(submissionConfig: Config): StreamJob = {
     info("Creating a ThreadJob, which is only meant for debugging.")
+    var config = submissionConfig
+    if (new JobConfig(submissionConfig).getConfigLoaderFactory.isPresent) {
```

Comment:
This block seems duplicated with `ProcessJobFactory`. Any benefit in sharing code?

Reply:
I am planning to simplify and refactor all of them after I introduces beam related planning.